tests/Hamcrest/Type/IsNumericTest.php

```diff
@@ -34,6 +35,10 @@ public function testEvaluatesToFalseIfArgumentDoesntMatchType()
         assertThat('foo', not(numericValue()));
         assertThat('foo5', not(numericValue()));
         assertThat('5foo', not(numericValue()));
+        assertThat('0x42A04G', not(numericValue())); // G is not in the hexadecimal range.
+        assertThat('0x4G2A04', not(numericValue())); // G is not in the hexadecimal range.
```

Comment:
Any reason for having 2 checks for `G` letter? I think, that `ctype_xdigit` is clever enough to detect it.